railties/lib/rails/generators/rails/app/templates/config/boot.rb

```diff
@@ -1,3 +1,17 @@
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
 
 require 'bundler/setup' # Set up gems listed in the Gemfile.
+
+if RUBY_ENGINE == 'ruby' && RUBY_PLATFORM =~ /darwin|linux|bsd/
```

Comment:
We should always require bootsnap and let it decide whether it supports the current environment.